packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/FunctionsTests.kt

```diff
@@ -0,0 +1,623 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// We use internal serialization APIs for testing purposes. No leaks to the public API.
+@file:OptIn(InternalSerializationApi::class)
+@file:Suppress("invisible_member", "invisible_reference")
+
+package io.realm.kotlin.test.mongodb.shared
+
+import io.realm.kotlin.internal.platform.runBlocking
+import io.realm.kotlin.internal.toMillis
+import io.realm.kotlin.internal.toRealmInstant
+import io.realm.kotlin.mongodb.Credentials
+import io.realm.kotlin.mongodb.Functions
+import io.realm.kotlin.mongodb.User
+import io.realm.kotlin.mongodb.call
+import io.realm.kotlin.mongodb.exceptions.FunctionExecutionException
+import io.realm.kotlin.mongodb.exceptions.ServiceException
+import io.realm.kotlin.test.assertFailsWithMessage
+import io.realm.kotlin.test.mongodb.TestApp
+import io.realm.kotlin.test.mongodb.createUserAndLogIn
+import io.realm.kotlin.test.mongodb.util.BaasApp
+import io.realm.kotlin.test.mongodb.util.Service
+import io.realm.kotlin.test.mongodb.util.TestAppInitializer.AUTHORIZED_ONLY_FUNCTION
+import io.realm.kotlin.test.mongodb.util.TestAppInitializer.ERROR_FUNCTION
+import io.realm.kotlin.test.mongodb.util.TestAppInitializer.FIRST_ARG_FUNCTION
+import io.realm.kotlin.test.mongodb.util.TestAppInitializer.NULL_FUNCTION
+import io.realm.kotlin.test.mongodb.util.TestAppInitializer.SUM_FUNCTION
+import io.realm.kotlin.test.mongodb.util.TestAppInitializer.VOID_FUNCTION
+import io.realm.kotlin.test.mongodb.util.TestAppInitializer.initializeDefault
+import io.realm.kotlin.types.RealmInstant
+import kotlinx.serialization.InternalSerializationApi
+import org.mongodb.kbson.BsonArray
+import org.mongodb.kbson.BsonBinary
+import org.mongodb.kbson.BsonBoolean
+import org.mongodb.kbson.BsonDecimal128
+import org.mongodb.kbson.BsonDocument
+import org.mongodb.kbson.BsonDouble
+import org.mongodb.kbson.BsonInt32
+import org.mongodb.kbson.BsonInt64
+import org.mongodb.kbson.BsonNull
+import org.mongodb.kbson.BsonString
+import org.mongodb.kbson.BsonType
+import org.mongodb.kbson.BsonUndefined
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertContentEquals
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+import kotlin.test.fail
+
+class FunctionsTests {
+//    @Serializable
+//    private data class Dog(var name: String? = null)
+
+//    private val looperThread = BlockingLooperThread()
```

Comment:
??

Reply:
The file is a copy-paste from Java, haven't trimmed any of the unused code. I was following what we do in other files, I prefer not to have these comments hanging out, we already have it in the Java project.